src/system/Form/Radio/Radio.tsx

```diff
@@ -0,0 +1,97 @@
+/** @jsxImportSource theme-ui */
+
+import classNames from 'classnames';
+import React, { Ref, forwardRef, useCallback } from 'react';
+import { ThemeUIStyleObject } from 'theme-ui';
+
+import { RadioOption } from './RadioOption';
+
+export const VIP_RADIO = 'vip-radio-component';
+
+export type RadioOptionOptionProps = {
+	id: string;
+	value: string;
+	disabled?: boolean;
+	className?: string;
+	label?: string;
+	renderLabel?: ( props, labelStyles: ThemeUIStyleObject ) => JSX.Element | null;
+	labelProps?: object;
+	inputProps?: object;
+};
+
+export interface RadioOptionProps {
+	option: RadioOptionOptionProps;
+	name: string;
+	variant: string;
+	disabled: boolean | undefined;
+	onChangeHandler: ( e: React.ChangeEvent< HTMLInputElement > ) => void;
+	checked: boolean;
+}
+
+export type RadioProps = {
+	variant?: string;
+	disabled?: boolean | undefined;
+	defaultValue?: string | number;
+	onChange?: ( e: React.ChangeEvent< HTMLInputElement >, option?: RadioOptionOptionProps ) => void;
+	name?: string;
+	options?: RadioOptionOptionProps[];
+	className?: string;
+};
+
+const Radio = forwardRef< HTMLDivElement, RadioProps >(
+	(
+		{
+			variant = 'primary',
+			disabled = false,
+			defaultValue,
+			onChange,
+			name = '',
+			options = [],
+			className,
+			...props
+		}: RadioProps,
+		ref: Ref< HTMLDivElement >
+	) => {
+		// If disabled is pass globally, it will overwrite the variant
+		if ( disabled === true || disabled === undefined ) {
+			variant = 'disabled';
+		}
+
+		const onChangeHandler = useCallback( ( e: React.ChangeEvent< HTMLInputElement > ) => {
```

Comment:
`options` are not passed to dependency array. if `options` change and component does not remount you have a bug. is `useCallback` really needed here?